lib/extensions/yaml_load_aliases.rb

```diff
@@ -0,0 +1,20 @@
+module YamlLoadAliases
+  # Psych 4 aliases load as safe_load.  Some loads happen early, like reading the database.yml so we don't want to load our
+  # constants at that time, such as MiqExpression, Ruport, so we have two sets of permitted classes.
+  def safe_load(yaml, permitted_classes: [], aliases: false, **kwargs)
+    # permitted_classes kwarg is provided because rails 6.1.7.x expects it as a defined kwarg.  See: https://github.com/rails/rails/blob/9ab33753b6bab1809fc73d35b98a5c1d0c96ba1b/activerecord/lib/active_record/coders/yaml_column.rb#L52
+    permitted_classes += YamlPermittedClasses.initialized? ? YamlPermittedClasses.app_yaml_permitted_classes : YamlPermittedClasses.default_permitted_classes
```

Comment:
there is a lot of text here.

Can we move this check into `YamlPermittedClasses`?

Something like:

```ruby
permitted_classes = YamlPermittedClasses.permitted_classes

class YmlPermittedClasses
  def self.permitted_classes
    YamlPermittedClasses.initialized? ? YamlPermittedClasses.app_yaml_permitted_classes : YamlPermittedClasses.default_permitted_classes
  end
end
```

napkin code - hopefully you can use for inspiration

Reply:
Thanks... now that I'm done looking at the yaml columns in other repos I'm going to come back to this.